### PR TITLE
Adding documentation for Trident RAS API

### PR DESCRIPTION
**Pull request by knusbaum**
Will merge to 1.x-branch as well.

### CODE REVIEW

**Comment by revans2** [comment]
+1 looks good to me.

File: docs/Trident-RAS-API.md [hunk 1]
```diff
@@ -0,0 +1,49 @@
+---
+title: Trident RAS API
+layout: documentation
+documentation: true
+---
+
+## Trident RAS API
+
+The Trident RAS (Resource Aware Scheduler) API provides a mechanism to specify the resource consumption of their topology. The API looks exactly like the base RAS API, only it is called on Trident Streams instead of Bolts and Spouts.
```
Comment (jerrypeng) [inline docs/Trident-RAS-API.md]: grammer:

... the resource consumption of **a Trident** topology ...

File: docs/Trident-RAS-API.md [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+
+In order to avoid duplication and inconsistency in documentation, the purpose and effects of resource setting are not described here, but are instead found in the [Resource Aware Scheduler Overview](Resource_Aware_Scheduler_overview.html)
+
+### Use
+
+First an example:
```
Comment (jerrypeng) [inline docs/Trident-RAS-API.md]: First,

File: docs/Trident-RAS-API.md [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+            .parallelismHint(10)
+            .setCPULoad(50)
+            .setMemoryLoad(1024)
+            .groupBy(new Fields("word!"))
+            .persistentAggregate(new MemoryMapState.Factory(), new Count(), new Fields("count"))
+            .setCPULoad(100)
+            .setMemoryLoad(2048);
+```
+
+Resources can be set per operation (except for grouping, shuffling, partitioning).
```
Comment (jerrypeng) [inline docs/Trident-RAS-API.md]: Resources can be set ** for each **operation

File: docs/Trident-RAS-API.md [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+Operations that are combined by Trident into single Bolts have their resources summed.
```
Comment (jerrypeng) [inline docs/Trident-RAS-API.md]: ... into single Bolts **will** have their ...

File: docs/Trident-RAS-API.md [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+
+Every Bolt is given **at least** the default resources, regardless of user settings.
+
+In the above case, we end up with
+ * a spout and spout coordinator with a CPU load of 20% each, and a memory load of 512MiB on heap and 256MiB off heap.
+ * a bolt with 60% cpu load (10% + 50%) and a memory load of 1536MiB (1024 + 512) on heap from the combined `Split` and `BangAdder`
+ * a bolt with 100% cpu load and a memory load of 2048MiB.
+
+The methods can be called for every operation (or some of the operations) or used in the same manner as `parallelismHint()`.
```
Comment (jerrypeng) [inline docs/Trident-RAS-API.md]: reword:
The RAS API can be used in the same manner as parrallelismHint(), i.e. resource declarations have the same *boundaries* as parallelismHints.

File: docs/Trident-RAS-API.md [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+            .setCPULoad(100)
+            .setMemoryLoad(2048);
+```
+
+Resources can be set per operation (except for grouping, shuffling, partitioning).
+Operations that are combined by Trident into single Bolts have their resources summed.
+
+Every Bolt is given **at least** the default resources, regardless of user settings.
+
+In the above case, we end up with
```
Comment (jerrypeng) [inline docs/Trident-RAS-API.md]: The aforementioned Trident Topology example will be transformed into a Storm Topology with the following components:

File: docs/Trident-RAS-API.md [hunk 7]
```diff
@@ -0,0 +1,50 @@
...
+
+Resources can be set for each operation (except for grouping, shuffling, partitioning).
+Operations that are combined by Trident into single Bolts will have their resources summed.
+
+Every Bolt is given **at least** the default resources, regardless of user settings.
+
+The Trident Stream above becomes a Storm Topology with:
+ * a spout and spout coordinator with a CPU load of 20% each, and a memory load of 512MiB on heap and 256MiB off heap.
+ * a bolt with 60% cpu load (10% + 50%) and a memory load of 1536MiB (1024 + 512) on heap from the combined `Split` and `BangAdder`
+ * a bolt with 100% cpu load and a memory load of 2048MiB.
```
Comment (satishd) [inline docs/Trident-RAS-API.md]: minor nit: 100% cpu load, and a memory load of 2048MiB **on heap and default value for off heap**

Comment (knusbaum) [inline docs/Trident-RAS-API.md]: Right you are.

**Comment by knusbaum** [comment]
Squashed and merged. Going in 1.x-branch as well.

**Pull request closed without merging**